alacritty_terminal/src/term/mod.rs

```diff
@@ -687,6 +687,83 @@ impl VisualBell {
     }
 }
 
+/// LineDamage describes the current damage of a line, represented as two
+/// column coordinates describing the edges of the damage. Damage is valid if
+/// left <= right.
+#[derive(Clone)]
+pub struct LineDamage {
+    pub left: Column,
+    pub right: Column,
+    pub line: Line,
+}
+
+impl LineDamage {
+    /// Create a new, undamaged LineDamage. where left=num_cols+1 and right=0.
+    /// This ensures that later update with min/max of any valid column
+    /// automatically creates a valid line damage without needing any
+    /// conditionals.
+    fn undamaged(line: Line, num_cols: Column) -> LineDamage {
+        LineDamage { left: num_cols + 1, right: Column(0), line }
+    }
+
+    /// Return the LineDamage to its undamaged state.
+    #[inline(always)]
+    pub fn reset(&mut self, num_cols: Column) {
+        self.left = num_cols + 1;
+        self.right = Column(0);
+    }
+
+    /// Return whether this line remains undamaged.
+    #[inline(always)]
+    pub fn is_undamaged(&self) -> bool {
+        self.left > self.right
+    }
+}
+
+/// Damage is used to track damage within a Term.
+pub struct Damage {
+    pub line_damage: Vec<LineDamage>,
+    pub damage_all: bool,
+    num_cols: Column,
+}
+
+impl Damage {
+    #[inline(always)]
+    fn new(lines: Line, num_cols: Column) -> Damage {
+        let mut damage =
+            Damage { line_damage: Vec::with_capacity(lines.0), damage_all: false, num_cols };
+        for line in 0..lines.0 {
+            damage.line_damage.push(LineDamage::undamaged(Line(line), num_cols));
+        }
+        damage
+    }
+
+    #[inline(always)]
+    fn expand_line_damage(&mut self, line: Line, left: Column, right: Column) {
+        let line = self.line_damage.get_mut(line.0).unwrap();
+        line.left = min(left, line.left);
+        line.right = max(right, line.right);
+    }
+
+    #[inline(always)]
+    fn clear_damage(&mut self) {
```

Comment:
I think it should be just `clear`, since Damage.clear_damage() is a bit of overuse of `damage`, what do you think? (same for `resize` and `expand_line_damage`, I guess)